libfuse/folderlist.go

```diff
@@ -161,26 +212,39 @@ func (fl *FolderList) Remove(ctx context.Context, req *fuse.RemoveRequest) (err
 	fl.fs.log.CDebugf(ctx, "FolderList Remove %s", req.Name)
 	defer func() { fl.fs.reportErr(ctx, libkbfs.WriteMode, err) }()
 
-	func() {
-		fl.mu.Lock()
-		defer fl.mu.Unlock()
-		if tlf, ok := fl.folders[req.Name]; ok {
-			// Fake future attr calls for this TLF until the user
-			// actually opens the TLF again, because some OSes (*cough
-			// OS X cough*) like to do immediate lookup/attr calls
-			// right after doing a remove, which would otherwise end
-			// up re-adding the favorite.
-			tlf.clearStoredDir()
-		}
-	}()
+	h, err := libkbfs.ParseTlfHandle(
```

Comment:
Ah, I missed that TODO.  Yes, `ParseTlfHandle` is the right way to do this.  I was worried this would slow down deletes, and make it hard to recover from bugs where we end up with non-canonical names in the favorites list for whatever reason.  But I guess it's the right thing to do, so I'm ok with it.